csrc/device_lower/analysis/sync_information.cpp

```diff
@@ -91,6 +91,34 @@ bool allowIncoherentDependency(
   return false;
 }
 
+// Check if an iter domain of a tensor is a subject of a scatter
+// op. Specifically, if the given expr is a scatter op using the given
+// tensor as its input, returns true if the given iter domain is
+// derived from the scattered logical iter domain.
+bool isConsumedByScatter(TensorView* tv, IterDomain* id, Expr* consumer_expr) {
+  auto scatter = dynamic_cast<ScatterOp*>(consumer_expr);
+  if (scatter == nullptr || scatter->in() != tv) {
+    return false;
+  }
+
+  auto logical_scatter_dim = tv->domain()->noReductions().at(scatter->dim());
```

Comment:
I'm not following here how we are using `tv->domain()->noReductions()` along with `scatter->dim()`.
Wasn't the first one the loop domain, while that scatter->dim() gives us index to the logical domain?